docs/parallel-node-execution.md

```diff
@@ -0,0 +1,203 @@
+## Overview
+
+Parallel node execution lets you run multiple AI agent nodes concurrently, improving performance and enabling complex workflows. This feature is particularly useful when you need to:
+
+- Process the same input through different models or approaches simultaneously
+- Perform multiple independent operations on the same data
```

Comment:
Maybe `Perform multiple independent operations in parallel` and don't mention `same data` here? It can associate with concurrent operation with agent's data, but we do it on copies, so no concurrent data sharing. Also, it could be both different data and different independent operations